### DISCUSSION TITLE
How to refresh a DataGrid manually

### DISCUSSION THREAD

**Discussion by mrkoenigstein** [vmware/clarity#5717] [Q&A] [answered]
How can I reload a DataGrid manually after I change some Data which is displayed with the Datagrid?

**Comment by bbogdanov** [comment]
Hi @cschulc,

Could you please give us more details about your scenario? 

We would appreciate if you could provide us a stackblitz example. 

Here are templates that you can use for the example `https://stackblitz.com/@clr-team`

**Reply by mrkoenigstein** [reply]
Hello Bogdan, it is difficult to provide an example, but I can try. 

My Scenario: I have a Datagrid with Reports and a Wizard to merge additional Result to a report. And after I merge extra Results to a Report I have to update the Datagrid because the Informations of the merged Report has changed. 
I need a way to use the refresh Methode which is explained [here](https://clarity.design/angular-components/datagrid/#server-driven-datagrid). But I don't know how to get the state: ClrDatagridStateInterface from the datagrid to call refresh.

Any Idea how to do this?

**Reply by hippee-lee** [reply]
Hi @cschulc - there is a public method `ClrDatagrid.dataChanged()` described [here](https://clarity.design/angular-components/datagrid/api/#clrdatagrid). You need to get a reference to the datagrid that has data changed and call the method as needed. It will trigger an update and refresh the datagrid.

**Reply by mrkoenigstein** [reply]
> Hi @cschulc - there is a public method `ClrDatagrid.dataChanged()` described [here](https://clarity.design/angular-components/datagrid/api/#clrdatagrid). You need to get a reference to the datagrid that has data changed and call the method as needed. It will trigger an update and refresh the datagrid.

Hi @hippee-lee, no the method dataChanged do not trigger the refesh method which is set to (clrDgRefresh) and the documetntation only mention "Re-trigger the computation of displayed items manually. (cell width and datagrid height)" so no refresh of data from the server.

**Reply by bbogdanov** [reply]
@cschulc https://stackblitz.com/edit/clarity-datagrid-add-new-data?file=src%2Fapp%2Fapp.component.html here is something I did based on your explanation. Feel free to fork and update it to get close to what you have as situation and lets find out what might be going wrong.

In the example as you can see the new data is been added from the wizard and the datagrid data is updated as intended.

**Comment by mrkoenigstein** [comment, marked as answer]
So a good way is the solution from @bbogdanov to push the new Item to the array of items, or to call refresh with the default state for the datagrid. Like this:

`  this.refresh({
      page: {
        current: 1,
        from: 0,
        size: 10,
        to: 9
      }
    });`
